test/integration/shoots/networking/shoot_networking_test.go

```diff
@@ -0,0 +1,179 @@
+		testLogger = logger.AddWriter(logger.NewLogger(*logLevel), GinkgoWriter)
+
+		var err error
+		shootGardenerTest, err = NewShootGardenerTest(*kubeconfig, nil, testLogger)
+		Expect(err).NotTo(HaveOccurred())
+
+		shoot := &gardencorev1alpha1.Shoot{ObjectMeta: metav1.ObjectMeta{Namespace: *shootNamespace, Name: *shootName}}
+		gardenerTestOperations, err = NewGardenTestOperationWithShoot(ctx, shootGardenerTest.GardenClient, testLogger, shoot)
+		Expect(err).NotTo(HaveOccurred())
+	}, InitializationTimeout)
+
+	CAfterSuite(func(ctx context.Context) {
+		err := gardenerTestOperations.ShootClient.Client().Delete(ctx, &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace}})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				Expect(err).To(HaveOccurred())
+			}
+		}
+	}, FinalizationTimeout)
+
+	CAfterEach(func(ctx context.Context) {
+		gardenerTestOperations.AfterEach(ctx)
+	}, DumpStateTimeout)
+
+	CIt("should reach all webservers on all nodes", func(ctx context.Context) {
+		templateFilepath := filepath.Join(resourcesDir, "templates", nginxTemplateName)
+		nettestTmpl := template.Must(template.ParseFiles(templateFilepath))
+
+		By("Deploy the net test daemon set")
+		var writer bytes.Buffer
+		err := nettestTmpl.Execute(&writer, map[string]string{
+			"name":      name,
+			"namespace": namespace,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = gardenerTestOperations.ShootClient.Applier().ApplyManifest(ctx, manifestReader, kubernetes.DefaultApplierOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = gardenerTestOperations.WaitUntilDaemonSetIsRunning(ctx, name, namespace, gardenerTestOperations.ShootClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		pods := &corev1.PodList{}
+		err = gardenerTestOperations.ShootClient.Client().List(ctx, pods, client.MatchingLabels{"app": "net-nginx"})
+		Expect(err).NotTo(HaveOccurred())
+
+		// check if all webservers can be reached from all nodes
+		By("test connectivity to webservers")
+		shootRESTConfig := gardenerTestOperations.ShootClient.RESTConfig()
+		var res error
+		for _, from := range pods.Items {
+			for _, to := range pods.Items {
+				By(fmt.Sprintf("Testing %s to %s", from.GetName(), to.GetName()))
+				reader, err := kubernetes.NewPodExecutor(shootRESTConfig).Execute(ctx, from.Namespace, from.Name, "net-curl", fmt.Sprintf("curl -L %s:80 --fail", to.Status.PodIP))
+				if err != nil {
+					res = multierror.Append(res, errors.Wrapf(err, "%s to %s", from.GetName(), to.GetName()))
+					continue
+				}
+				data, err := ioutil.ReadAll(reader)
+				if err != nil {
+					testLogger.Error(err)
+					continue
+				}
+				testLogger.Infof("%s to %s: %s", from.GetName(), to.GetName(), data)
+			}
+		}
+		Expect(res).ToNot(HaveOccurred())
+
+		err = gardenerTestOperations.ShootClient.Client().Delete(ctx, &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace}})
```

Comment:
Is that needed although it's already deleted via `CAfterSuite`?

Reply:
I wanted the cleanup to also be part of the test. But you're right this is only needed in the afterSuite 👍